cuaca/api.py

```diff
@@ -1,63 +1,49 @@
+"""Python wrapper for the Malaysian Weather Service API.
+
+To use this, first register an account at https://api.met.gov.my/
+(I wish this public project is really public)
+"""
 import os
 import datetime
 import pickle
 import requests
 
+
 class WeatherAPI(object):
-    """
-    This is a python wrapper for the Malaysian Weather Service API. 
-    TO use this, first register an account(I wish this public project is really public)
+    """Weather API wrapper.
 
-    url at https://api.met.gov.my/
+    >>> from cuaca
+    >>> api = cuaca.api.WeatherAPI("API_KEY")
     """
 
     def __init__(self, api_key, end_point="https://api.met.gov.my/v2/", cache_dir=None):
-        """
-        api = WeatherAPI("API_KEY")
-        """
         self.end_point = end_point
-        self.headers = {
-                "Authorization" : "METToken %s" % api_key
-        }
-        self.cache_dir = cache_dir
-        self.cache, self.cache_index = {}, {}
+        self.headers = {"Authorization": "METToken %s" % api_key}
         self.cache_expiry = datetime.timedelta(days=1)
-        if self.cache_dir:
-            self.load_cache()
-
-    def cache_expire(self):
-        now = datetime.datetime.now()
-        for k, v in self.cache_index.items():
-            if now - k >= self.cache_expiry:
-                for c in v:
-                    try:
-                        del self.cache[c]
-                    except:
-                        pass
-
-    def load_cache(self):
-        try:
-            with open(os.path.join(self.cache_dir, "cuaca.p"), "rb") as fp:
-                self.cache = pickle.load(fp)
-        except FileNotFoundError:
-                pass
-        try:
-            with open(os.path.join(self.cache_dir, "cuacaidx.p"), "rb") as fp:
-                self.cache_index = pickle.load(fp)
-            self.cache_expire()
-        except FileNotFoundError:
-            pass
-
-    def save_cache(self):
-        if not self.cache_dir: return
-        self.cache_expire()
-        try:
-            with open(os.path.join(self.cache_dir, "cuaca.p"), "wb") as fp:
-                pickle.dump(self.cache, fp)
-            with open(os.path.join(self.cache_dir, "cuacaidx.p"), "wb") as fp:
-                pickle.dump(self.cache_index, fp)
-        except:
-            raise
+        if cache_dir:
+            self.cache_file = os.path.join(cache_dir, "cuaca.p")
+            self.cache = self._load_cache()
+        else:
+            self.cache = {}
+
+    def __del__(self):  # XXX: Should we implement __exit__?
```

Comment:
keep this for now. But the reason why we don't put save cache here is, we do not always want cache to be generate. The usecase for this is for testing, we want to see if the API is working, and we do not want the test result is wrong due to old cache.